### PR TITLE
Upgrade Vitessce to 1.1.0 (Shareable Links + Scale bar for Microscopy)

### LINKED ISSUE #1005: add scale bar to imaging data displays
Scale bars should be activated in Vitessce for all image data.

### LINKED ISSUE #1065: Save State in Vitessce
Just want to create something here to track https://github.com/hubmapconsortium/vitessce/issues/241

### LINKED ISSUE #1259: Implement Vitessce Shareable Link
See #1250 for design. You should currently have view access to the Figma, but let me know if you need edit access.

### PR DESCRIPTION
**Pull request by ilan-gold**
This fixes #1259, fixes #1005, and fixes #1065.

I think only one of your reviews should be enough, but just wanted to give a heads up so you could test it out.  I think the length of the URL should not be a problem, regardless of the browser.

### CODE REVIEW

**Review by keller-mark — CHANGES_REQUESTED** [review]
Looks good. I think the final Cell Type Annotation preview view config is missing `'version': '0.1.0'` (HBM396.RPRR.624)

**Comment by ilan-gold** [comment]
@keller-mark Good catch.  Just fixed it.  Also addressed Nils' comments

**Review by keller-mark — APPROVED** [review]

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:62]: Possibly pull this out to a hooks or utils file?

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:76]: This isn't keyboard accessible. I think there are a few props you can pass to the mui menu components to handle this.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:74]: We should probably not use in-line styles, but not a blocking issue.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:52]: Rename `setEmailUrl` or `encodeEmailUrl` or similar?

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:37]: Possibly pull these functions out to a different file?

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (john-conroy) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:79]: Was use callback needed here? Is it because of the vitessce internals?

**Review by john-conroy — APPROVED** [review]

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (john-conroy) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:74]: I know we kept in-line styles on similar popper components, this is more so a note to myself.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:52]: I think if you're going with the "verb first" convention, that's what I did here which I can see might not be clear.  Maybe `createEmailWithUrl` since this function actually opens your email app.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:74]: Yes, this is why I did so here.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:64]:
```suggestion
    const fragment = window.location.hash.substr(1);
    const vitessceURLConf = fragment && decodeURLParamsToConf(fragment);
```

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:72]: This whole chunk can probably be simpler... hard to understand what it's doing right now.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:79]: add a comment? what goes wrong without the debounce?

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:154]:
```suggestion
            message={`Shareable URL copied to clipboard. ${onCopyUrlWarning}`}
```

File: context/app/static/js/components/Detail/visualization/Visualization/style.js
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/style.js:51]: Is there an existing palette color to use? if not, let's add this to the global palette... Might loop in @tsliaw .

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:21]:
```suggestion
  'Warning: Long URLs may not work on some browsers.';
```

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:30]:
```suggestion

function VisualizationShareButton() {
```

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:45]:
```suggestion
    const encodedConf = encodeConfInUrl({
      conf,
      onOverMaximumUrlLength: () => {
        setOnCopyUrlWarning(DEFAULT_LONG_URL_WARNING);
      },
    });
    const url = `${window.location.href.split('#')[0]}#${encodedConf}`;
```

Too much nesting gets hard to read.

Comment (mccalluc) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:45]: Actually, pull this out as a utility function, since it's copy and pasted below.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:62]:
```suggestion
    const mailtoLink = `mailto:?body=${longUrlWarning}${encodeURI('\r\n\r\n')}${DEFAULT_EMAIL_MESSAGE} ${encodedUrl}`;
```

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:74]: move to style.js?

**Review by mccalluc — CHANGES_REQUESTED** [review]
Lots of little comments. Also, could you add some unit tests? The useMemo logic is kind of complicated... I think it can be simplified, but a test would help too.

Would it be feasible to add either react component tests, or something in Cypress?

**Comment by john-conroy** [comment]
From standup: Does it make sense to only set the config when it's needed instead of using the on change?

We don't think it's a big issue, and the pathway to doing so might introduce more issues than it is worth.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (john-conroy) [reply context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx]: From standup: Does it make sense to move the useCallback into vitessce? Possibly, but probably fine as is.

Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx]: Not needed (or rather, not performant).  Has been removed.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:44]: Discussed offline - will be resolved in a separate PR.

Comment (john-conroy) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:44]: Some of our other menus which follow this pattern aren't keyboard accessible either, so this isn't an issue specific to this PR. I'll just file an issue and do this later. Thanks!

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/VisualizationShareButton.jsx:42]: Will be resolved in future PR.

File: context/app/static/js/components/Detail/visualization/Visualization/hooks.js
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/hooks.js:22]: Would it make sense to separate the `isMultiDataset` case, rather than having repeated checks? Not sure.

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/constants.js
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/VisualizationShareButton/constants.js:5]: Not sure we gain that much by splitting this out.

**Review by mccalluc — COMMENTED** [review]
Did a little bit more manual QA, and didn't notice any problems. Realize that I haven't used the multidataset functionality... that's in the Showcases?

I have a stylistic preference for boolean short circuits over `?:` ternaries, particularly when one branch is boolean-ish. Not a big deal, but it can make it a little bit harder to read... Like, if the "else" is `null`, someone reading the code from the top has to find the end of the "then" to understand what's going on.

I'm a little worried about the performance for people with slightly older browsers (FF, in particular)... particularly because it's a degradation, and not an outright error, it can be hard for the user to know what the problem is, and they'll either leave in frustration, or inn the best case, we get a really unhelpful bug report. Maybe something like the terms-of-use click-through would be appropriate? ... but maybe that could be a separate PR?

**Review by mccalluc — COMMENTED** [review]
Sorry, meant the last review to be a "comment"... I think there should be a few more tweaks, and I think there should be some consideration of messaging for folks with older browsers, but I don't need to re-review before you merge. Thanks!

**Comment by ilan-gold** [comment]
@mccalluc We should discuss the FF use case during the UI meeting today.  Otherwise I'll address these comments.  Additionally, the CODEX datasets are all multi-dataset.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:158]: This is a "false dif" so this is the same as before.  Not sure if @john-conroy or @keller-mark implmented this....

File: context/app/static/js/components/Detail/visualization/VisualizationShareButton/constants.js
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/VisualizationShareButton/constants.js:5]: I'm in general a fan of "constants" files especially because the number is now going to go up again if we show a message for FF.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:166]: I think there is probably some sort of internal check for truthiness here so I am going to leave this.  I am in general a fan of ternaries because they are clear about what they do.  Especially in jsx, you can end up rendering things unintentionally with && or || if you are not careful.  This change would have broken full-screen mode.

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (ilan-gold) [reply context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:100]: I'll add a comment - it's meant to prevent multi datasets from rendering before the selection has been initialized.

**Comment by ilan-gold** [comment]
@mccalluc Could you look at the [warning for FF](https://github.com/hubmapconsortium/portal-ui/pull/1371/files#diff-e1a15accd7a4a60ccb99335eb4deda2d28b5c9ceceeeeda2512832abaa38c718R38-R40)?

File: context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx
Comment (mccalluc) [inline context/app/static/js/components/Detail/visualization/Visualization/Visualization.jsx:39]:
```suggestion
  'On older versions of Firefox, zooming in on visualizations is very slow; Upgrading to the latest version will fix the problem.';
```
- "may be" -> "is"
- Be more more precise about what interactions are slow.
- A little shorter.

Thanks!

**Review by mccalluc — COMMENTED** [review]
Thanks! This is how I might phrase it, but your call.

**Pull request merged**
